### ISSUE TITLE
duplicate items while doing REPORT request by sharee

### ISSUE THREAD

**Issue by nabim777** [owncloud/ocis#7028]
## Describe the bug

for the all `dav-path-version`, when Sharee searches for a resource then duplicate response cames.

## Steps to reproduce

1. As `marie`, create two folder `folder1` and `folder2`
2. As `marie`, upload a file `file1.txt` in `folder1`
3. As `marie`, upload a file `file2.txt` in `folder2`
4. As `marie`, share both `folder1` and `folder2` with `einstein`
5. As `einstein`, accept the share request
6. As `einstein`, search resources using `file1` through API

Search command:
```bash
curl -kv -XREPORT https://localhost:9200/remote.php/dav/spaces/<einstein-personal-spaceId> \
-u einstein:relativity \
-d "<?xml version='1.0' encoding='utf-8' ?>
<oc:search-files xmlns:a='DAV:' xmlns:oc='http://owncloud.org/ns' >
<oc:search>
<oc:pattern>file1</oc:pattern>
</oc:search>
</oc:search-files>"
```

## Expected behavior

<details>

<summary>The response should contain only one item</summary>

```xml
< HTTP/1.1 207 Multi-Status
...
<?xml version="1.0"?>
<d:multistatus xmlns:s="http://sabredav.org/ns" xmlns:d="DAV:" xmlns:oc="http://owncloud.org/ns">
  <d:response>
    <d:href>/remote.php/dav/spaces/a0ca6a90-a365-4782-871e-d44447bbc668%21dd862bea-f1b8-47b4-8c78-70c859209fdf:8c72645c-8f24-4ae3-a7d7-3bd16a52f946:13aa78ab-289d-47a2-bc05-1578d6014247/folder1/file1.txt</d:href>
    <d:propstat>
      <d:prop>
        <oc:fileid>dd862bea-f1b8-47b4-8c78-70c859209fdf$8c72645c-8f24-4ae3-a7d7-3bd16a52f946!66dd399f-34fe-46fe-b01f-2f85eb42142a</oc:fileid>
        <oc:file-parent>dd862bea-f1b8-47b4-8c78-70c859209fdf$8c72645c-8f24-4ae3-a7d7-3bd16a52f946!5c900653-f977-464a-8450-c124dfc6bccc</oc:file-parent>
        <oc:shareid>dd862bea-f1b8-47b4-8c78-70c859209fdf:8c72645c-8f24-4ae3-a7d7-3bd16a52f946:d8e2311b-7b87-434d-8b8c-b3773322f58f</oc:shareid>
        <oc:shareroot>/folder1</oc:shareroot>
        <oc:name>file1.txt</oc:name>
        <d:getlastmodified>2023-08-14T05:56:46Z</d:getlastmodified>
        <d:getcontenttype>text/plain</d:getcontenttype>
        <oc:permissions>S</oc:permissions>
        <oc:highlights/>
        <oc:tags/>
        <d:getetag/>
        <d:resourcetype/>
        <d:getcontentlength>14</d:getcontentlength>
        <oc:score>9.16186809539795</oc:score>
      </d:prop>
      <d:status>HTTP/1.1 200 OK</d:status>
    </d:propstat>
  </d:response>
```

</details>

## Actual behavior

share root is different but file name is same
As we are searching for `file1.txt` but in the response `file1.txt` is displayed twice with share root

this is not true 
 ```xml
  <oc:shareroot>/folder2</oc:shareroot>
```
but this below is true
```xml
<oc:shareroot>/folder1</oc:shareroot>
```

<details>
<summary>duplicate items with different shareroot</summary>

```xml
< HTTP/1.1 207 Multi-Status
...
<?xml version="1.0"?>
<d:multistatus xmlns:s="http://sabredav.org/ns" xmlns:d="DAV:" xmlns:oc="http://owncloud.org/ns">
  <d:response>
    <d:href>/remote.php/dav/spaces/a0ca6a90-a365-4782-871e-d44447bbc668%21dd862bea-f1b8-47b4-8c78-70c859209fdf:8c72645c-8f24-4ae3-a7d7-3bd16a52f946:d8e2311b-7b87-434d-8b8c-b3773322f58f/file1.txt</d:href>
    <d:propstat>
      <d:prop>
        <oc:fileid>dd862bea-f1b8-47b4-8c78-70c859209fdf$8c72645c-8f24-4ae3-a7d7-3bd16a52f946!66dd399f-34fe-46fe-b01f-2f85eb42142a</oc:fileid>
        <oc:file-parent>dd862bea-f1b8-47b4-8c78-70c859209fdf$8c72645c-8f24-4ae3-a7d7-3bd16a52f946!5c900653-f977-464a-8450-c124dfc6bccc</oc:file-parent>
        <oc:shareid>dd862bea-f1b8-47b4-8c78-70c859209fdf:8c72645c-8f24-4ae3-a7d7-3bd16a52f946:d8e2311b-7b87-434d-8b8c-b3773322f58f</oc:shareid>
        <oc:shareroot>/folder1</oc:shareroot>
        <oc:name>file1.txt</oc:name>
        <d:getlastmodified>2023-08-14T05:56:46Z</d:getlastmodified>
        <d:getcontenttype>text/plain</d:getcontenttype>
        <oc:permissions>S</oc:permissions>
        <oc:highlights/>
        <oc:tags/>
        <d:getetag/>
        <d:resourcetype/>
        <d:getcontentlength>14</d:getcontentlength>
        <oc:score>9.16186809539795</oc:score>
      </d:prop>
      <d:status>HTTP/1.1 200 OK</d:status>
    </d:propstat>
  </d:response>
  <d:response>
    <d:href>/remote.php/dav/spaces/a0ca6a90-a365-4782-871e-d44447bbc668%21dd862bea-f1b8-47b4-8c78-70c859209fdf:8c72645c-8f24-4ae3-a7d7-3bd16a52f946:13aa78ab-289d-47a2-bc05-1578d6014247/folder1/file1.txt</d:href>
    <d:propstat>
      <d:prop>
        <oc:fileid>dd862bea-f1b8-47b4-8c78-70c859209fdf$8c72645c-8f24-4ae3-a7d7-3bd16a52f946!66dd399f-34fe-46fe-b01f-2f85eb42142a</oc:fileid>
        <oc:file-parent>dd862bea-f1b8-47b4-8c78-70c859209fdf$8c72645c-8f24-4ae3-a7d7-3bd16a52f946!5c900653-f977-464a-8450-c124dfc6bccc</oc:file-parent>
        <oc:shareid>dd862bea-f1b8-47b4-8c78-70c859209fdf:8c72645c-8f24-4ae3-a7d7-3bd16a52f946:13aa78ab-289d-47a2-bc05-1578d6014247</oc:shareid>
        <oc:shareroot>/folder2</oc:shareroot>
        <oc:name>file1.txt</oc:name>
        <d:getlastmodified>2023-08-14T05:56:46Z</d:getlastmodified>
        <d:getcontenttype>text/plain</d:getcontenttype>
        <oc:permissions>S</oc:permissions>
        <oc:highlights/>
        <oc:tags/>
        <d:getetag/>
        <d:resourcetype/>
        <d:getcontentlength>14</d:getcontentlength>
        <oc:score>9.16186809539795</oc:score>
      </d:prop>
      <d:status>HTTP/1.1 200 OK</d:status>
    </d:propstat>
  </d:response>
</d:multistatus>
```

</details>

**Comment by JammingBen** [comment]
I noticed the same today. @2403905 fyi

**Comment by SwikritiT** [comment]
this issue and https://github.com/owncloud/ocis/issues/7092 might be related

**Comment by SwikritiT** [comment]
hi @2403905 and @nabim777 does the fix for https://github.com/owncloud/ocis/issues/7092 fix this issue too? Maybe we can close this issue as well?

**Comment by nabim777** [comment]
> hi @2403905 and @nabim777 does the fix for #7092 fix this issue too? Maybe we can close this issue as well?

Due to this https://github.com/owncloud/ocis/issues/7092 fix issue the duplicate items are not reproduced now
